# Title

## Summary

## Introduction

## Methods and Results

To start, we will import the required libraries for our analysis, set the random state to generate reproducible results, and read in the data.

In [1]:
# import required libraries for analysis
import altair as alt
import pandas as pd
import numpy as np
import sklearn
from sklearn.model_selection import train_test_split
from sklearn.dummy import DummyClassifier
from sklearn.model_selection import train_test_split, cross_validate, GridSearchCV, RandomizedSearchCV
from sklearn.preprocessing import (FunctionTransformer, Normalizer, OneHotEncoder, StandardScaler, normalize, scale)
from sklearn.compose import make_column_transformer
from sklearn.metrics import make_scorer
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score
from sklearn.pipeline import Pipeline, make_pipeline
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import  confusion_matrix, ConfusionMatrixDisplay, classification_report

# set random state to have reproducible results
random_state=12

# read in data
raw_full_flight_data = pd.read_csv("../data/raw/full_data_flightdelay.csv")

Let's see how big the data set is.

In [2]:
raw_full_flight_data.shape

(6489062, 26)

There are 6,489,062 observations in the raw data set. Since such a large data set will take a lot of computing power and time, we will take sample of the data (20,000 observations) and use it in our analysis.

In [3]:
# sample 20,000 observations from the raw data set
raw_sample_flight_data = raw_full_flight_data.sample(n=20000, random_state=12)

# save the sample data set
raw_sample_flight_data.to_csv("../data/processed/raw_sample_flight_data.csv")

# check shape of sample to confirm the sampling worked
raw_sample_flight_data.shape

(20000, 26)

Let's clean our sample data by only keeping the features of interest and the target column in our data. Then, we'll split the data into training and testing sets.

In [4]:
# list of features and target (DEP_DEL15) columns
list_of_features_and_target = ['MONTH', 'DAY_OF_WEEK', 'DEP_DEL15', 'CONCURRENT_FLIGHTS', 'CARRIER_NAME',
 'FLT_ATTENDANTS_PER_PASS', 'GROUND_SERV_PER_PASS', 'PLANE_AGE',
 'DEPARTING_AIRPORT', 'PREVIOUS_AIRPORT', 'SNOW', 'AWND']

# only keep the features of interest and the target column in the data set
filtered_sample_flight_data = raw_sample_flight_data[list_of_features_and_target]

# save the filtered sample data set
filtered_sample_flight_data.to_csv("../data/processed/filtered_sample_flight_data.csv")

# split filtered sample data into training and testing splits.
flight_train, flight_test = train_test_split(filtered_sample_flight_data, test_size=0.2, random_state=12, stratify=filtered_sample_flight_data["DEP_DEL15"])

# save taining and test splits.
flight_train.to_csv("../data/processed/training_flight_data.csv")
flight_test.to_csv("../data/processed/testing_flight_data.csv")

## Discussion

## References